jupyterhub/auth.py

```diff
@@ -210,6 +209,57 @@ def validate_username(self, username):
         """
     )
 
+    post_auth_hook = Any(config=True,
+        help="""
+        An optional hook function that you can implement to do some
+        bootstrapping work during authentication. For example, loading user account
+        details from an external system.
+        
+        This function is called after the user has passed all authentication checks
+        and is ready to successfully authenticate. This function must return the
+        authentication dict reguardless of changes to it.
+        
+        This maybe a coroutine.
+
+        Example::
+
+            import os, pwd
+            def my_hook(authenticator, handler, authentication):
+                user_data = pwd.getpwnam(authentication['name'])
+                spawn_data = {
+                    'pw_data': user_data
+                    'gid_list': os.getgrouplist(authentication['name'], user_data.pw_gid)
+                }
+                
+                if authentication['auth_state'] is None:
```

Comment:
Do you think we can change the default `auth_state` from None to an empty dict without too many headaches Fumbling with None makes the example more complex.